furusiya/model/entities/game_object.py

```diff
@@ -18,30 +18,36 @@ def __init__(self, x, y, char, name, color, blocks=False):
 
         self._components = {}
 
-    def _get_type(self, type_):
-        possible_matches = [
+    def _get_components_of_type(self, type_):
+        return [
             k
             for k in self._components.keys()
-            if k == type_.component_type
+            if issubclass(k, type_)
```

Comment:
I think we differed over this, the last thing I remember is that we said we just want to return where the type is exact, and let consumers of the ECS figure out how they want to deal with subclasses.

I'm on Discord now if you want to chat about this in more detail.